### PR TITLE
Resolve task properties only once

### PR DESCRIPTION
**Pull request by lptr**

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/DefaultTaskProperties.java [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+
+        return new DefaultTaskProperties(
+            task.toString(),
+            inputPropertiesVisitor.getPropertyValuesFactory().create(),
+            inputFilesVisitor.getFileProperties(),
+            outputFilesVisitor.getFileProperties(),
+            localStateVisitor.getFiles(),
+            destroyablesVisitor.getFiles(),
+            validationVisitor.getTaskPropertySpecs(),
+            validationVisitor.getMessages());
```
Comment (wolfs) [inline subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/DefaultTaskProperties.java]: Messages is now always empty.

Comment (lptr) [reply subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/DefaultTaskProperties.java]: Thanks, removed.

**Comment by lptr** [comment]
Branch build: https://builds.gradle.org/viewQueued.html?itemId=9895089

**Comment by lptr** [comment]
Looks like we are breaking some serious tests :/

**Comment by wolfs** [comment]
That means that the following needs to be done lazily, too: https://github.com/gradle/gradle/blob/6ba7332a917c8c2f25aa4e14a832b9c45a7ec1d3/subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/GetOutputFilesVisitor.java#L39-L50

**Comment by lptr** [comment]
Why?

**Comment by wolfs** [comment]
I thought this was a problem I already had before. But it seems to be something new.
Seems to me like some input properties are evaluated too early.

**Comment by wolfs** [comment]
I guess we shouldn't resolve input properties at all for building the dependency graph, only input files.

**Comment by lptr** [comment]
Build again: https://builds.gradle.org/viewQueued.html?itemId=9897480

**Comment by lptr** [comment]
Next: https://builds.gradle.org/viewQueued.html?itemId=9897903

**Comment by lptr** [comment]
Build: https://builds.gradle.org/viewQueued.html?itemId=9899452

**Pull request merged**